### ISSUE TITLE
Issue: New .NET Foundation Project Application

### ISSUE THREAD

**Issue by TonyTroeff** [dotnet-foundation/projects#333]
### Project Name

TryAtSoftware.Extensions.Reflection

### License

MIT

### Contributor

TonyTroeff

### Existing OSS Project?

Yes

### Source Code URL

https://github.com/TryAtSoftware/Extensions

### Project Transfer Signatories

- Tonyslav Troeff <tonyslav_troeff@outlook.com>

### Description

`TryAtSoftware.Extensions.Reflection` is a library containing extension methods and utility components that should simplify (and optimize) some common operations with reflection.

### Name

Tonyslav Troeff

### Email

tonyslav_troeff@outlook.com

### GitHub Profile URL

https://github.com/TonyTroeff

### Governance Model

In order to propose new changes, the repository should be forked and a Pull Request must be created (from the forked repository) to the original one. The whole process is described within the `CONTRIBUTING.md` file.

### CLA

- [ ] If already an OSS project, was a Contribution License Agreement in place for contributions accepted?

### How does the project check who has signed one?

I am willing to make the required changes in order to introduce CLA.

### Select the Project Transfer Agreement model

Contribution

### Repository Layout

Currently, within this repository, there are multiple projects from the `TryAtSoftware.Extensions.*` family. If necessary, each one of them may be separated.

### Eligibility Criteria

- [X] The project is built on the .NET platform and/or creates value within the .NET ecosystem.
- [X] The project produces source code for distribution to the public at no charge.
- [X] The project's code is easily discoverable and publicly accessible (preferably on GitHub).
- [X] The project contains a build script that can produce deployable artifacts that are identical to the official deployable artifacts, with the exception of code signing (Exception may be granted for strong name keys, though strongly encouraged to be committed. Exception relies on OSS signing being in the build script for public builds).
- [X] When applicable, project must use reproducible build settings in its toolchain.
- [X] The project uses [Source Link.](https://docs.microsoft.com/en-us/dotnet/standard/library-guidance/sourcelink)
- [X] The project uses either embedded PDBs or publish symbol packages to NuGet (if applicable).
- [ ] The project code signs their artifacts as appropriate.
- [X] The project organization has 2FA enabled. Requiring 2FA must be done as part of onboarding if not already enabled.
- [X] Libraries that are mandatory dependencies of the project are offered under a standard, permissive open source license which has been approved by the .NET Foundation (exceptions include a dependency that is required by the target platform where no alternative open source dependency is available such as the .NET Framework or a hardware specific library).
- [X] Committers are bound by a Contributor License Agreement (CLA) and/or are willing to embrace the .NET Foundation's CLA when the project becomes a Member.
- [X] The copyright ownership of everything that the project produces is clearly defined and documented.
- [X] The project has a public issue tracker where the status of any defect can be easily obtained.
- [X] The project has a published Security Policy.
- [X] The project has a home page which provides high level information about its status and purpose.
- [X] The project has a public communication channel where community members can engage with maintainers.
- [X] The project has a publicly available location where members can review and contribute to documentation.

### Infrastructure Requirements Summary

There are no requirements - the projects can be consumed as a NuGet package.

**Comment by sbwalker** [comment]
@TonyTroeff can you please clarify if you are applying to have your entire open source project https://github.com/TryAtSoftware/Extensions join the .NET Foundation... or if you are only submitting the TryAtSoftware.Extensions.Reflection class.

**Comment by TonyTroeff** [comment]
I do not know which would be easier. Potentially, I would like to apply with all of them. However, they may consist of multiple separate NuGet packages (and there are more yet to come). I would appreciate your advices.

**Comment by sbwalker** [comment]
Usually an application includes all of the IP which is grouped under the same open source license. So in your case it appears that the entire repo including all of your extensions is offered under the same MIT-based permissive license. So your application should include the entire repo. If you wanted to only contribute a specific Extension then you would need to manage that extension as an independent piece of IP under its own distinct license. However I don't think that is your intent in this situation. Note that it does not matter if you may have 1 or more Nuget packages depending on your deployment strategy. What matters is that all items in your distribution are licensed under the same open source license. I hope this makes sense.

**Comment by TonyTroeff** [comment]
Yes, it does, In that case, it makes sense to include everything that is included within the repository. Should I create a new application or edit this one?

**Comment by sbwalker** [comment]
I would suggest simply editing this application

**Comment by TonyTroeff** [comment]
Done!
Excuse me for the delayed response. I was sick the last few days...

**Comment by sbwalker** [comment]
@TonyTroeff the Project Committee reviewed your application and it was noted that the Windows Community Toolkit (https://github.com/CommunityToolkit/WindowsCommunityToolkit) is a "utility" project that is already a member of the .NET Foundation - so perhaps there may be an option to approach its maintainer(s) to determine if there is interest in joining forces. From a maintainers point of view, I completely understand if this does not align with your personal goals for the project... but the Committee thought it would be a good idea to raise the question regardless. Please share your thoughts so that we can determine next steps for your application.

**Comment by TonyTroeff** [comment]
Hello!

From what I know and have seen, the Windows Community Toolkit is mostly focused on the development of UI components. However, the idea of `TryAtSoftware.Extensions` is to expose extension methods and utility components simplifying common tasks at lower or higher level of abstraction and complexity. It is platform-independent and, moreover, all libraries are developed under .NET Standard.
The two projects just look too different to me.
